security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/token/oauth2/OAuth2DataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,215 @@
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.security.spring.authentication.token.oauth2;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+
+import org.cbioportal.model.DataAccessToken;
+import org.cbioportal.service.DataAccessTokenService;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.jwt.Jwt;
+import org.springframework.security.jwt.JwtHelper;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+
+@Service
+@Component("oauth2")
+public class OAuth2DataAccessTokenServiceImpl implements DataAccessTokenService {
+
+    @Value("${dat.oauth2.issuer:}")
+    private String issuer;
+
+    @Value("${dat.oauth2.clientId:}")
+    private String clientId;
+
+    @Value("${dat.oauth2.clientSecret:}")
+    private String clientSecret;
+
+    @Value("${dat.oauth2.accessTokenUri:}")
+    private String accessTokenUri;
+
+    @Value("${dat.oauth2.userAuthorizationUri:}")
+    private String userAuthorizationUri;
+
+    @Value("${dat.oauth2.redirectUri:}")
```

Comment:
Setting this is missing from the documents.

Reply:
Fixed.